forc-pkg/src/manifest.rs

```diff
@@ -150,6 +153,18 @@ impl Manifest {
             Dependency::Simple(_) => None,
         })
     }
+
+    /// Parse and return the associated project's program type.
+    pub fn program_type(&self, manifest_dir: PathBuf) -> Result<TreeType> {
+        let manifest = Manifest::from_dir(&manifest_dir)?;
```

Comment:
Ahh I missed this previously - there's no need to load a new `Manifest` in this method, we should use `self` instead.

Reply:
Like so? 

```pub fn program_type(&self, manifest_dir: PathBuf) -> Result<TreeType> {
        let entry_string = self.entry_string(&manifest_dir)?;
        let program_type = parse(entry_string, None);

        match program_type.value {
            Some(parse_tree) => Ok(parse_tree.tree_type),
            None => bail!(parsing_failed(&self.project.name, program_type.errors)),
        }
    }
```